model/task/task.go

```diff
@@ -2471,83 +2472,101 @@ func (t *Task) Insert() error {
 // into the old_tasks collection. If this is a display task, its execution tasks
 // are also archived.
 func (t *Task) Archive() error {
-	if t.DisplayOnly && len(t.ExecutionTasks) > 0 {
-		execTasks, err := FindAll(db.Query(ByIds(t.ExecutionTasks)))
+	if t.DisplayOnly {
+		var task []Task
+		task = append(task, *t)
+		err := ArchiveMany(task)
+		return errors.Wrapf(err, "archiving and updating display task with ID '%s'.", t.Id)
+	} else {
+		// Archiving a single task.
+		archiveTask := t.makeArchivedTask()
+		err := db.Insert(OldCollection, archiveTask)
 		if err != nil {
-			return errors.Wrap(err, "retrieving execution tasks")
+			grip.Error(message.WrapError(err, message.Fields{
+				"archive_task_id": archiveTask.Id,
+				"old_task_id":     archiveTask.OldTaskId,
+				"execution":       t.Execution,
+				"display_only":    t.DisplayOnly,
+			}))
+			return errors.Wrap(err, "inserting archived task into old tasks")
 		}
-		if err = ArchiveMany(execTasks); err != nil {
-			return errors.Wrap(err, "archiving execution tasks")
+		err = UpdateOne(
+			bson.M{IdKey: t.Id},
+			bson.M{
+				"$unset": bson.M{
+					AbortedKey:              "",
+					AbortInfoKey:            "",
+					OverrideDependenciesKey: "",
+				},
+				"$set": bson.M{ExecutionKey: t.Execution + 1, LatestParentExecutionKey: t.Execution + 1},
```

Comment:
For tasks, their execution and parent execution should be the same. Since this isn't using the pipeline code, it doesn't have the set of setting latest parent execution to the conditional: if latest parent execution exists, then latest parent execution += 1 else, latest parent execution = execution + 1.

Essentially, it is using execution for backwards compatability and using the fact that they should be equal always for tasks.